frappe/integrations/doctype/google_calendar/google_calendar.py

```diff
@@ -683,55 +612,79 @@ def get_recurrence_parameters(recurrence):
-				'sequence': 1,
-				'reminders': {
-					'useDefault': True
+def get_event_attendees(event):
+	event_participants = []
+	for item in event.get("ATTENDEE", []):
+		if item.startswith("mailto:"):
+			attendee_email = item[len("mailto:") :]
+			if frappe.db.exists("User", {"email": attendee_email}):
+				event_participants.append(
+					{
+						"reference_doctype": "User",
+						"reference_docname": frappe.get_doc("User", {"email": attendee_email}).name,
+					}
+				)
+			elif frappe.db.exists("Contact", {"email_id": attendee_email}):
+				event_participants.append(
+					{
+						"reference_doctype": "Contact",
+						"reference_docname": frappe.get_doc("Contact", {"email_id": attendee_email}).name,
+					}
+				)
+			else:
+				contact = {
+					"doctype": "Contact",
+					"first_name": attendee_email.split("@")[0],
+					"email_id": attendee_email,
+					"email_ids": [{"email_id": attendee_email, "is_primary": 1}],
 				}
-			}
-		]
-	}
-	*recurrence
-		- Daily Event: ['RRULE:FREQ=DAILY']
-		- Weekly Event: ['RRULE:FREQ=WEEKLY;BYDAY=MO,TU,TH']
-		- Monthly Event: ['RRULE:FREQ=MONTHLY;BYDAY=4TH']
-			- BYDAY: -2, -1, 1, 2, 3, 4 with weekdays (-2 edge case for April 2017 had 6 weeks in a month)
-		- Yearly Event: ['RRULE:FREQ=YEARLY;']
-		- Custom Event: ['RRULE:FREQ=WEEKLY;WKST=SU;UNTIL=20191028;BYDAY=MO,WE']"""
+				doc = frappe.get_doc(contact).insert(ignore_permissions=True)
+				event_participants.append({"reference_doctype": "Contact", "reference_docname": doc.name})
+	return event_participants
+
+
+def get_participant_email(guest):
+	doc = frappe.get_doc(guest.get("reference_doctype"), guest.get("reference_docname"))
+	return doc.get("email") if guest.get("reference_doctype") == "User" else doc.get("email_id")
+
+
+def create_calendar_object(account, doc):
+	caldav_url = (
+		"https://apidata.googleusercontent.com/caldav/v2/{}/events/{}?access_token={}".format(
+			account.user, doc.name, account.get_access_token()
+		)
+	)
+	cal = Calendar()
+	cal["PRODID"] = "-//Google Inc//Google Calendar 70.9054//EN"
+	cal["VERSION"] = "2.0"
+	cal["CALSCALE"] = account.user
+	cal["X-WR-CALNAME"] = account.user
+	cal["X-WR-TIMEZONE"] = frappe.db.get_single_value("System Settings", "time_zone")
+	return caldav_url, cal
+
+
+def enqueue_job(job, **kwargs):
+	check_scheduler_status()
+	account = kwargs.get("account") or {}
+	job_name = "calendar_import|{}".format(account.get("name"))
+	if not job_already_enqueued(job_name):
+		enqueue(
+			job,
+			**kwargs,
+			queue="long",
+			timeout=10000,
+			event="Import events from google ",
+			job_name=job_name,
+			now=frappe.conf.developer_mode or frappe.flags.in_test
+		)
+
+
+def check_scheduler_status():
+	if is_scheduler_inactive() and not frappe.flags.in_test:
+		frappe.throw(_("Scheduler is inactive. Cannot enqueue job."), title=_("Scheduler Inactive"))
+
+
+def job_already_enqueued(job_name):
+	enqueued_jobs = [d.get("job_name") for d in get_info()]
+	if job_name in enqueued_jobs:
+		return True
```

Comment:
Logic for the last few utils can be simplified. Haven't checked the general flow yet though.